crates/holo_hash/src/hash_ext.rs

```diff
@@ -88,3 +97,59 @@ fn hash_from_content<T: HashType, C: HashableContent<HashType = T>>(content: &C)
         HashableContentBytes::Prehashed39(bytes) => HoloHash::from_raw_39_panicky(bytes),
     }
 }
+
+/// Adds convenience methods for constructing HoloHash and HoloHashed
+/// from some HashableContent
+pub trait HashableContentExtSync<T>: HashableContent
```

Comment:
While benchmarking for the tokio+libsodium-based sodoken library, I determined the optimal switchover for blake2b from sync to async was ~ `50` KiB of data to hash... assuming the blake2b_simd library we're using here has similar performance:

- what's the size limit for header / entry content? is it below this, and we might not even need an async version?
- if not, do we still have warnings on the sync version if we go over this?

Reply:
Header size is around 100 bytes, up to whatever our limit on LinkTag is, which is currently 1KB, so yeah that's under.

Max entry size is 16MB so we could asyncify that. How much do we gain though, at the expense of the inconvenience of making more places in the code async? Are the gains worth it?

Reply:
We used to have a panic when the size of sync-hashed content was over 16MB, but that seems to have been removed somewhere down the line... https://github.com/holochain/holochain/blob/562af8f7a286ea30dfbce564a6b2607401d4527c/crates/holo_hash/src/hash_ext.rs#L16-L26

Reply:
It's not about gains... it about not having code that "blocks" our tokio core threads resulting in task starvation.

Reply:
Yeah but we calculated that even on a HoloPort nano, blake2b hashing 16MB is well under 20ms, and I thought being under 20ms between awaits was the goal. You mentioned an optimal switchover, but my questioning is aiming at finding out how steep the slope of that performance curve actually is (and if it extends to more complex system behavior).